src/LogikUI/Simulation/Gates/AndGate.cs

```diff
@@ -15,14 +15,9 @@ struct AndGateInstance : IInstance, IEquatable<AndGateInstance>
         public Vector2i Position { get; set; }
         public Orientation Orientation { get; set; }
         public int NumberOfPorts => 3;
+        public UIntPtr BackendIdx { get; set; }
```

Comment:
In the latest c# pr the way we store components have completely changed.
So we will have to decide which pr to merge first and who has to fix the marge conflicts.
I think we should merge my pr first as it already has an id field in the new data structure so there will be less code to edit overall as when merging we can just omit this code.